### PR TITLE
MINIFICPP-1091 - Update LICENSE and NOTICE

### PR DESCRIPTION
**Pull request by bakaid**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -2531,3 +2531,101 @@ This product bundles 'zlib' within 'OpenCV' under the following license:
...
+would be appreciated but is not required.
+
+2. Altered source versions must be plainly marked as such, and
+must not be misrepresented as being the original software.
+
+3. This notice may not be removed or altered from any source
+distribution.
+
+
+This product bundles 'cJSON' within 'AWS SDK for C++' under an MIT license:
```
Comment (bakaid) [inline LICENSE]: The `NOTICE.txt` of aws-sdk-cpp says that they bundle JsonCpp, but this is no longer true, as evidenced by the code and the `1.5.0` part of https://github.com/aws/aws-sdk-cpp/blob/master/CHANGELOG.md
They now bundle cJSON, so I have included the license for that here.

File: LICENSE [hunk 2]
```diff
@@ -2531,3 +2531,101 @@ This product bundles 'zlib' within 'OpenCV' under the following license:
...
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+
+This product bundles 'Android tool chain cmake build files' within 'AWS SDK for C++' under an MIT license:
```
Comment (bakaid) [inline LICENSE]: There is no explicit license file for this in the aws-sdk-cpp repo, just a mention in https://github.com/aws/aws-sdk-cpp/blob/66666d9437429c28927576cb5c898e490f8c8cf9/LICENSE.txt#L57 that it is an MIT license and the copyright holders, so I have added an MIT license here.

File: LICENSE [hunk 3]
```diff
@@ -203,6 +203,12 @@
 
 APACHE NIFI - MINIFI SUBCOMPONENTS:
 
+The Apache NiFi - MiNiFi project contains subcomponents licensed under the Apache License, Version 2.0:
```
Comment (arpadboda) [inline LICENSE]: I know it's just moved, but my gut feeling says that we should call the project MiNiFi C++ here as well to distinguish cpp and java versions. 

@apiri what's your opinion?

**Comment by apiri** [comment]
reviewing

File: LICENSE (hunk 3 above)
Comment (apiri) [reply LICENSE]: yeah, that seems fair.  it was originally done as such from the standpoint that it was under a common effort/subproject but for licensing clarity, clearly differentiating will be helpful

File: LICENSE [hunk 4]
```diff
@@ -1793,3 +1620,1012 @@ ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
...
+(https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html).
+
+========================================================================
+tinyDTLS
+
+When compiled with tinyDTLS support, this software includes components
+that are licensed under the terms of the Eclipse Distribution License 1.0
+(http://www.eclipse.org/org/documents/edl-v10.php).
+
+This project bundles 'open62541' which is available under an MPLv2.0 license,
```
Comment (apiri) [inline LICENSE]: It is my understanding this is a binary inclusion.  We should be explicit about that fact as it is a weak copyleft license and its use outside of that capacity would not be okay.  I see the conditional nature of it is highlighted in the README, but we should be also update the language here.

File: LICENSE (hunk 3 above)
Comment (bakaid) [reply LICENSE]: Agreed, will update all instances.

File: LICENSE (hunk 4 above)
Comment (bakaid) [reply LICENSE]: Yep, it is a binary-only inclusion, will update the language to reflect this.

**Comment by bakaid** [comment]
@apiri @arpadboda thanks for the reviews, addressed the review comments in a new commit.

**Pull request closed without merging**